### PR TITLE
fix(touch): enhance parseOptions and fix touch's -r flag

### PR DESCRIPTION
**Pull request by nfischer**
This improves `parseOptions` to allow options of the form:

```
touch({'-r', 'path/to/file'}, 'other-file.txt');
```

to supply a value to an option. In this example, `-r` has the string value `'path/to/file'`.

### CODE REVIEW

File: src/common.js [hunk 1]
```diff
@@ -80,33 +82,44 @@ function parseOptions(str, map) {
...
-      else
-        options[opt] = true;
-    } else {
-      error('option not recognized: '+c);
-    }
-  });
+  var optionName;
+  if (typeof opt === 'string') {
+    // e.g. match[1] = 'Rf' for opt = '-Rf'
+    var match = opt.match(/^\-(.+)/);
```
Comment (ariporad) [inline src/common.js]: This would be much more performant if it just was `opt[0] === '-'`.

**Comment by ariporad** [comment]
One small nit, otherwise looks good!

**Comment by ariporad** [comment]
@nfischer: In the changelog, how should this be described?

**Comment by nfischer** [comment]
@ariporad I replaced the instances of `match()` with suitable alternatives

**Comment by ariporad** [comment]
@nfischer: Fantastic!

**Comment by nfischer** [comment]
In the changelog, you can just say something like "feat: now supports valued options (see `touch -r` for an example)."

**Comment by ariporad** [comment]
Will Do!

**Comment by ariporad** [comment]
LGTM!

**Pull request merged**